x/chainlink/keeper/keeper.go

```diff
@@ -372,23 +386,37 @@ func (k Keeper) DistributeReward(ctx sdk.Context, submitter sdk.AccAddress, data
 
 	// distribute reward to all data providers except submitter
 	for _, dp := range dataProviders {
-		if dp.Address.String() != submitter.String() {
+		if dp.Address.String() != msg.Submitter.String() {
 			if err := k.bankKeeper.SendCoinsFromModuleToAccount(
 				ctx, types.ModuleName, dp.Address, sdk.NewCoins(tokensToSend),
 			); err != nil {
 				return err
 			}
+			// emit OraclePaid event for valid data providers
+			OraclePaidEvent.Account = dp.Address
+			err := types.EmitEvent(OraclePaidEvent, ctx.EventManager())
+			if err != nil {
+				return err
+			}
 		}
 	}
 
 	// send to submitter
 	// TODO: include fees - need to mint this amount as well
 	if err := k.bankKeeper.SendCoinsFromModuleToAccount(
-		ctx, types.ModuleName, submitter, sdk.NewCoins(tokensToSend),
+		ctx, types.ModuleName, msg.Submitter, sdk.NewCoins(tokensToSend),
 	); err != nil {
 		return err
 	}
 
+	// emit OraclePaid event to submitter including the fee
+	OraclePaidEvent.Account = msg.Submitter
```

Comment:
```suggestion
	OraclePaidEvent.Account = msg.Submitter
	OraclePaidEvent.Value = uint64(feedReward)
```

Reply:
this is probably the right way because the submitter reward will be slightly more than the dp's rewards because of the additional fee+gas compensation. so we should separate the `Value` for the two events emitted